Sources/SwiftBasicFormat/Trivia+FormatExtensions.swift

```diff
@@ -0,0 +1,99 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+
+extension Trivia {
+  /// Removes all blanks that is trailing before a newline trivia,
+  /// effectively making sure that lines don't end with a blank
+  func trimingTrailingBlanksBeforeNewline(isBeforeNewline: Bool) -> Trivia {
```

Comment:
This is really nice! 
Looking forward to how many bytes we can save in Swift Syntax 😆